### PR TITLE
Base Tags base functionality. 

### PR DESCRIPTION
**Pull request by zyuhel**
according to issue #100

### CODE REVIEW

**Comment by zyuhel** [comment]
:)  Tests still missed, will add test for youtube and couple of others soon.

File: src/Providers/OEmbed.php [hunk 1]
```diff
@@ -104,6 +104,23 @@ public function getType()
...
      */
+    public function getTags()
+    {
+        if ($this->bag->has('meta')) {
+            //it means we are using iframe.ly api
+            $meta_bag=$this->bag->get('meta');
+            if (isset($meta_bag['keywords'])) {
+                 return array_map('trim', explode(',',$meta_bag['keywords']));
+            }
+        }
```
Comment (oscarotero) [inline src/Providers/OEmbed.php]: FYI, you can use directly `$this->bag->has('meta[keywords]')` and `$this->bag->get('meta[keywords]')` (shorter and simpler)

File: src/Utils.php [hunk 2]
```diff
@@ -176,6 +172,28 @@ public static function getFirstValue(array $values, $returnKey = false)
...
+
+    /**
+     * Returns values as array
+     *
+     * @param array $values    The array provided by self::getData()
+     * @param bool  $returnKey Whether or not return the key instead the value
+     *
+     * @return array
+     */
+    public static function getArrayValue(array $values, $returnKey = false)
```
Comment (oscarotero) [inline src/Utils.php]: Maybe, a better name for this function is `getAllValues`, more understandable, what do you think?

Comment (zyuhel) [inline src/Utils.php]: yes, getAllValues seems more related. will commit fix.

File: src/Providers/OEmbed.php (hunk 1 above)
Comment (zyuhel) [inline src/Providers/OEmbed.php]: thanks will use shorter one.

**Comment by oscarotero** [comment]
Great! Can I merge this? Is it finished?

**Comment by zyuhel** [comment]
Yes some more test need to be added , may be writing extended support for different adapters. But in general it works.

**Comment by oscarotero** [comment]
Fantastic work. thank you!

**Pull request merged**